.travis.yml

```diff
@@ -85,110 +54,41 @@ cache:
     - node_modules
     - $HOME/.atom
     - $HOME/.stack
+    - $HOME/.opam
 
 before_install:
-  # Install Homebrew on Linux
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
-      git clone --depth=1 https://github.com/Linuxbrew/brew.git ~/.linuxbrew || true;
-    fi
-  # Update Homebrew
-  - brew update
-  - brew tap homebrew/dupes
-  - brew tap homebrew/versions
-  # Ruby language support
-  # - gem install ruby-beautify --verbose
-  # - gem install rubocop
-  - gem install htmlbeautifier
-  - gem install puppet-lint
-  # Sass language support
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
-      gem install sass;
-    else
-      docker pull unibeautify/sass-convert;
-    fi
-  # Python language support
+    # Linux: Install Nix and Packages, Python install
+    # OSX: Update Homebrew, remove conflicting cask, brew bundle, symlink elm-format
+    #      Python install
   - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
-      sudo chmod 777 -R /opt/python;
-      pip install --upgrade pip;
-      pip install --user --upgrade autopep8;
-      pip install --user --upgrade isort;
+      curl https://nixos.org/nix/install | sh &&
+      . ~/.nix-profile/etc/profile.d/nix.sh &&
+      nix-env -i uncrustify R elm-format terraform atom atom-beta stack &&
+      pip install --upgrade pip &&
+      pip install --user -r requirements.txt;
+    elif [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+      brew update && brew cask zap oclint && brew bundle &&
+      ln -s /usr/local/bin/elm-format-0.17 /usr/local/bin/elm-format &&
+      pip2 install --upgrade pip &&
+      pip2 install -r requirements.txt;
     else
-      pip2 install --upgrade pip2;
-      pip2 install --upgrade autopep8;
-      pip2 install --upgrade isort;
-    fi
-  # SQL language support
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
-      pip install --user --upgrade sqlparse;
-    else
-      pip2 install --upgrade sqlparse;
-    fi
-  # Java, C, C++, C#, Objective-C, D, Pawn, Vala
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
-      brew install uncrustify;
-    else
-      docker pull unibeautify/uncrustify;
+      echo Error:TRAVIS_OS_NAME && exit 1;
```

Comment:
I think this should be:

```bash
echo "Error: $TRAVIS_OS_NAME" &&  && exit 1;
```

Reply:
why the double `&&`? i can't get that to work

Reply:
Yea don't need the double ampersands, but need `"Error: $TRAVIS_OS_NAME"` in quotes.

Reply:
@szeck87 still invalid `yaml` https://travis-ci.org/Glavin001/atom-beautify/builds/327070475

Reply:
Yea that's odd, I thought you had to put quotes when echoing a string.  Sorry about that.  That line probably won't ever be hit, at least in the near future, as the only OS names are linux and osx.

Reply:
Oops. Typo on my end. Thanks for correcting!